### PR TITLE
Upload suggestions to global

### PR DESCRIPTION
**Pull request by antopalidi**
fixes stadtluzern/decidim-pdf-consultations/issues/37

Screenshot 2023-06-13 at 13 20 15

Screenshot 2023-06-13 at 13 21 53

Screenshot 2023-06-13 at 14 09 36

Screenshot 2023-06-13 at 14 31 03

### CODE REVIEW

File: app/forms/decidim/participatory_documents/suggestion_form.rb [hunk 1]
```diff
@@ -3,8 +3,38 @@
...
+
+      def validate_body_length
+        return if body.blank?
+
+        if body.length < MIN_BODY_LENGTH
+          errors.add(:base, I18n.t("activemodel.errors.models.suggestion.attributes.too_short", min_length: MIN_BODY_LENGTH))
+        elsif body.length > MAX_BODY_LENGTH
+          errors.add(:base, I18n.t("activemodel.errors.models.suggestion.attributes.too_long", max_length: MAX_BODY_LENGTH))
+        end
+      end
```
Comment (microstudi) [inline app/forms/decidim/participatory_documents/suggestion_form.rb:37]: May it be that we need test cases for these conditions?

**Review by microstudi — COMMENTED** [review]
There's one thing I would change from the UI,
currently we have this:

Screenshot from 2023-06-12 23-38-48

In my view, the current design occupies too much space and I think it could be solved in a more compact way. I also think that the main use case for the "upload file" is going to be to complement the text, not to do one think or the other (sorry we talk about this and we decided the other way) but I think we were wrong and this would make much more sense (design could be improved):

image

In this case, you don't just "upload a file", you attach document to the text. And, when pressing the "Attach file" text, the dialog for uploading the file would just appear. No instructions are needed (anyway, the current instructions have no meaning, it says something about images in landscape and CSVs).
For a premium behaviour, we could even accept drag & drop over the textarea to upload the file (but that's not necessary now).

When a file has been uploaded, we could show the file uploaded instead of the link to attach a file.

**Comment by antopalidi** [comment]
@microstudi I'm done, PR is ready for review

File: app/forms/decidim/participatory_documents/suggestion_form.rb [hunk 2]
```diff
@@ -3,8 +3,33 @@
 module Decidim
   module ParticipatoryDocuments
     class SuggestionForm < Decidim::Form
+      include Decidim::AttachmentAttributes
+      include Decidim::HasUploadValidations
+
+      MIN_BODY_LENGTH = 15
+      MAX_BODY_LENGTH = 150
```
Comment (microstudi) [inline app/forms/decidim/participatory_documents/suggestion_form.rb:10]: We need to make this configurable, as a `config_accessor` at least in the plugin (maybe also as a component configuration option). Pointing this here to not forget about it but probable to be done in another PR.

**Review by microstudi — APPROVED** [review]
great work!
